core/ArchiveProcessor/Loader.php

```diff
@@ -144,24 +205,14 @@ private function prepareArchiveImpl($pluginName)
         // visits archive can be inaccurate in the long run.
         if ($this->canSkipThisArchive()) {
             if (!empty($idArchives)) {
+                $this->quickReturn = true;
                 return [$idArchives, $visits];
             } else {
+                $this->quickReturn = true;
                 return [false, 0];
             }
         }
-
-        if (SettingsServer::isArchivePhpTriggered()) {
-            $this->logger->info("initiating archiving via core:archive for " . $this->params);
-        }
-
-        list($visits, $visitsConverted) = $this->prepareCoreMetricsArchive($visits, $visitsConverted);
-        list($idArchive, $visits) = $this->prepareAllPluginsArchive($visits, $visitsConverted);
-
-        if ($this->isThereSomeVisits($visits) || PluginsArchiver::doesAnyPluginArchiveWithoutVisits()) {
-            return [[$idArchive], $visits];
-        }
-
-        return [false, false];
+        return [$visits, $visitsConverted];
```

Comment:
If I see that correctly, this return is actually the only case where `quickReturn` would be `false` and the return values are actually not used, as the archiving will be triggered, right?
Guess you could simply return `null` or `false` in that case and check for that in the main method to get rid of `quickReturn` at all maybe.